llvm/lib/Target/AMDGPU/Disassembler/AMDGPUDisassembler.cpp

```diff
@@ -803,6 +803,21 @@ DecodeStatus AMDGPUDisassembler::getInstruction(MCInst &MI, uint64_t &Size,
     }
   }
 
+  // Validate buffer offset for GFX12+ - must be positive
+  if ((MCII->get(MI.getOpcode()).TSFlags &
+       (SIInstrFlags::MTBUF | SIInstrFlags::MUBUF)) &&
+      AMDGPU::isGFX12Plus(STI)) {
```

Comment:
Avoid generation checks

Reply:
The property is not gfx12+, the property is this specific opcode supports the offset or not.